src/api/product/product.ts

```diff
@@ -13,3 +16,20 @@ export const createGetProductsUrl = (params: GetProductsUrlParams): string => {
 
   return `machines?${searchParams.toString()}`;
 };
+
+export const validateProductsUrlParams = (query: DecodedValueMap<QueryParamConfigMap>) => {
+  const validateNumber = (number: number | undefined, defaultNumber: number | undefined) =>
+    number && Number.isInteger(number) && number > 0 ? number : defaultNumber;
+
+  const validatePage = validateNumber(query.page, defaultProductParams.page);
+  const validatePerPage = validateNumber(query.perPage, defaultProductParams.perPage);
+  const validateCategory = Array.isArray(query.category)
```

Comment:
Te trzy nazwy powyżej wskazują, że to są funkcje. Czy to są funkcje?

Reply:
Słusznie, nie są. Powinno być validated